src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -504,17 +500,38 @@ private static void RemoveDirectoryInternal(DirectoryInfo directory, bool recurs
                     case Interop.Error.EPERM:
                     case Interop.Error.EROFS:
                     case Interop.Error.EISDIR:
-                        throw new IOException(SR.Format(SR.UnauthorizedAccess_IODenied_Path, directory.FullName)); // match Win32 exception
+                        throw new IOException(SR.Format(SR.UnauthorizedAccess_IODenied_Path, fullPath)); // match Win32 exception
                     case Interop.Error.ENOENT:
-                        if (!throwOnTopLevelDirectoryNotFound)
+                        if (!topLevel)
                         {
-                            return;
+                            return true;
+                        }
```

Comment:
This one's confusing me, so I'll try to explain the logic. Please correct me if I got it wrong:

The [rmdir docs](https://man7.org/linux/man-pages/man2/rmdir.2.html) say:

```
ENOENT A directory component in pathname does not exist or is a dangling symbolic link.
```

So if we get ENOENT, it means the current entry is a symlink. And only if the current entry is the top level, we avoid deleting the contents of the link _target_ recursively. We achieve this by returning `true`, to assume the operation succedeed.

Was that correct?


But then, why do we throw (by going to default) if the current entry is not the top level?

Reply:
> ENOENT A directory component in pathname does not exist or is a dangling symbolic link.

You should read this as:

> ENOENT A directory component in pathname does not exist or _A directory component in pathname_ is a dangling symbolic link.

When the full pathname is a symbolic link, dangling or not, `rmdir` returns `ENOTDIR`.

Reply:
I've added a comment.